src/useQueryParam.ts

```diff
@@ -30,19 +29,7 @@ export const useQueryParam = <D, D2 = D>(
   paramConfig: QueryParamConfig<D, D2> = StringParam as QueryParamConfig<any>,
   rawQuery?: EncodedQueryWithNulls
 ): [D2 | undefined, (newValue: D, updateType?: UrlUpdateType) => void] => {
-  const { history, location } = React.useContext(QueryParamContext);
-
-  // ref with current version history object (see #46)
-  const refHistory = React.useRef(history);
```

Comment:
I understand that refLocation is necessary to stop the callback updating itself repeatedly, but I'm not sure why refHistory is needed? Doesn't it stay the same throughout the app?

Reply:
When I swap the order like you mention, I get an error about the child of <Router> needs to have certain props (i forget what it says now), so that's why I switched the order. Maybe my reach router example dependency is out of date though.

Reply:
Perhaps QueryParamsProvider needed some reach router specific props when rendering under Router. Anyway, the history-caching changes you've made fix this bug too.